### ISSUE TITLE
`$$.display()` updates dropped if cell execution completed

### ISSUE THREAD

**Issue by n-riesco** [n-riesco/ijavascript#132]
```js
$$.async();

var counter = $$.display("counter");

var n = 0;
counter.text(n++);

(function($$) {
    var stopCounter = (function() {
        var id = setInterval(function() {
            counter.text(n++);
        }, 1000);
        return function() {
            console.log("done");
            clearInterval(id);
            $$.done();
        }
    })();
    setTimeout(stopCounter, 10000);    
})($$);
```

**Comment by n-riesco** [comment]
```js
var d1 =  $$.display("d1");
d1.text("1");
d1.text("2");
```
jupyter-notebook@4.2.3 final output is 1
jupyter-notebook@5.2.1 final output is 2

**Comment by n-riesco** [comment]
```py
import os
from IPython.display import display
handle = display(0, display_id="counter")
for n in range(1, 10):
    os.system("sleep 1")
    handle.update(n)
```
jupyter-notebook@4.2.3 thrown expection at `handle.update(n)`
jupyter-notebook@5.2.1 works as expected

**Comment by n-riesco** [comment]
https://github.com/n-riesco/nel/commit/0322589c978a664bdffecdfe0a1f2f698afc02fc addresses this issue (needs testing with `jupyter-notebook@4.2.3` and `jupyter-notebook@5.2.1`).

**Comment by n-riesco** [comment]
https://github.com/n-riesco/nel/commit/0322589c978a664bdffecdfe0a1f2f698afc02fc tested on `jupyter-notebook@5.2.1` using the example:

```js
var counter = $$.display("counter");

var n = 0;
counter.text('' + n++);

(function($$) {
    var stopCounter = (function() {
        var id = setInterval(function() {
            counter.text('' + n++);
        }, 1000);
        return function() {
            console.log("done");
            clearInterval(id);
        }
    })();
    setTimeout(stopCounter, 10000);    
})($$);
```

---

Note the following gotchas:
- jupyter-notebook@4.2.3 doesn't support `update_display_data` messages
- the jupyter notebook doesn't show `display_data` messages sent before the execution result
- Note that I had to change the example in my first comment:
  - removed `$$.async()` and `$$.done()` to ensure `display_data` messages are sent after the execution result.
  - replaced `counter.text(n++);` by `counter.text('' + n++);` to ensure a string is sent

**Comment by n-riesco** [comment]
Closed via [NEL@0.5.8](https://github.com/n-riesco/nel/commit/4fae76657cab35547b1b3b5592c216d71dc34448)
